### PR TITLE
Feat/pvp  ranked rewards on burnt chars

### PR DESCRIPTION
**Pull request by Jcalcaldev**
This PR:

-Fixes an issue with restartRankedSeason if one of the top 3 players leaves the arena and burns it's character.
-Updates shield minting in tests (broken since last week).
-Should add some gas saving in some functions by calling _ownerByCharacter instead of characters.ownerOf

### CODE REVIEW

**Comment by poshdan** [comment]
Shouldn't isOwnedCharacter also use the new _ownerByCharacter value?

**Comment by Jcalcaldev** [comment]
> Shouldn't isOwnedCharacter also use the new _ownerByCharacter value?

This is optimal but would cause some characters to be trapped in an invalid state in current ongoing networks (like BSC Testnet). Since _ownerByCharacter is set upon entering arena, players who have their characters already inside would not be able to leave and re-enter. I'm not sure how to approach this issue in the most optimal way.

**Comment by poshdan** [comment]
It should be possible to expel players via an admin call.
The only concern there is gas, but on OEC it's very negligible.
We shouldn't be making compromises when a little elbow grease can help bridge the gap

File: contracts/PvpArena.sol [hunk 1]
```diff
@@ -118,6 +118,8 @@ contract PvpArena is Initializable, AccessControlUpgradeable {
     uint256 public duelOffsetCost;
     address payable public pvpBotAddress;
 
+    /// @dev owner's address by character ID
+    mapping(uint256 => address) private _ownerByCharacter;
     mapping(uint256 => uint256) public specialWeaponRerollTimestamp;
```
Comment (poshdan) [inline contracts/PvpArena.sol:123]: the new mapping must come after the last state variable, please correct

**Review by poshdan — CHANGES_REQUESTED** [review]
one small change needed afaik

**Review by kcper — APPROVED** [review]

**Review by poshdan — APPROVED** [review]

**Pull request merged**